haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
-        self._sock.setblocking(True)
-        self._sock.settimeout(self.connection._connect_timeout)
-        if self.connection._sock_opts:
-            for k, v in self.connection._sock_opts.iteritems():
-                family, type = k
-                self._sock.setsockopt(family, type, v)
-        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
-        exc = socket.error("getaddrinfo returns an empty list")
-        for _family, _socktype, _proto, _canonname, sockaddr in infos:
+
+        for info in socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP):
+
+            family, socktype, proto, _, sockaddr = info
+            self._sock = klass(family, socktype, proto)
```

Comment:
@kevinconway: Creation and configuration of a socket apparently can also fail. We should catch `socket.error`, log it, and recover just like we do for the `sock.connect` case. See "echo client program" example below "The next two examples are identical to the above two, but support both IPv4 and IPv6" in https://docs.python.org/2/library/socket.html. This is also demonstrated in my example PR #80


It should be something like this:

```
        try:
            self._sock = klass(family, socktype, proto)
            self._sock.settimeout(self.connection._connect_timeout)
            if self.connection._sock_opts:
                _sock_opts = self.connection._sock_opts
                for (level, optname), value in _sock_opts.iteritems():
                    self._sock.setsockopt(level, optname, value)
        except socket.error:
            self.connection.logger.exception(
                'Failed to create or configure socket; '
                'addr_info=%s; sock_opts=%s",
                info, sock_opts)
            continue
```